eng/targets/CSharp.Common.targets

```
@@ -16,6 +16,12 @@
     <!-- Public members should not use oblivious types. Not done with all nullable annotations. -->
     <NoWarn>$(NoWarn);RS0041</NoWarn>
 
+    <Nullable Condition="'$(Nullable)' == '' AND !('$(IsTestProject)' == 'true' OR
+        '$(IsTestAssetProject)' == 'true' OR
+        '$(ISBenchmarkProject)' == 'true' OR
+        '$(IsSampleProject)' == 'true' OR
+        '$(IsMicrobenchmarksProject)' == 'true')">enable</Nullable>
```

This rhymes with
```suggestion
    <Nullable Condition=" '$(Nullable)' == '' AND (
        '$(IsImplementationProject)' == 'true' OR
        '$(IsAnalyzersProject)' == true OR
        '$(IsSpecificationTestProject)' == 'true') ">enable</Nullable>
```
If that's what you meant, the suggestion is easier to read. Even better if analyzers and / or specification tests don't need nullability enabled.

---

Separately, may be good to add bits like `AND ! $(RepoRelativeProjectDir.Contains('Tools')` rather than enshrine exclusions we'll never remove elsewhere in the repo.

---

I didn't quite follow the second suggestion. Is this to exclude the eng tools?

---

The second suggestion would remove the need for src/Tools/Directory.Build.props because `$(RepoRelativeProjectDir)` is the relative path from the repo root to the project folder.

/fyi src/ProjectTemplates/testassets/DotNetToolsInstaller is the only not-exactly-`Tools` folder the `Contains(...)` would match. I suspect a repetition w/ `.Contains("tools")` would be needed if you want to automatically exclude projects under src/Middleware/tools and the other 10 or so other `tools` folders.

---

I really want to get to the tools too (it validates the nullability of shared sources for e.g). Just not making it a priority for 6.0. Having it be an explicit nullable disabled makes it easier to find these exclusions. Let me know if you feel strongly about it.